File: lib_java/src/main/java/lib/kalu/adapter/manager/CrashGridLayoutManager.java
package lib.kalu.adapter.manager;

import android.content.Context;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.AttributeSet;

/**
 * description: fix-bug-IndexOutOfBoundsException: Inconsistency detected. Invalid view holder adapter的解决方案
 * created by kalu on 2017/3/24 15:05
 */
public final class CrashGridLayoutManager extends GridLayoutManager {

    private final String TAG = CrashGridLayoutManager.class.getSimpleName();

    public CrashGridLayoutManager(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    public CrashGridLayoutManager(Context context, int spanCount) {
        super(context, spanCount);
    }

    public CrashGridLayoutManager(Context context, int spanCount, int orientation, boolean reverseLayout) {
        super(context, spanCount, orientation, reverseLayout);
    }

    @Override
    public void onLayoutChildren(RecyclerView.Recycler recycler, RecyclerView.State state) {
        try {
            super.onLayoutChildren(recycler, state);
        } catch (Exception e) {
         //   LogUtil.e(TAG, e.getMessage(), e);
        }
    }
}

File: lib_kotlin/src/main/java/lib/kalu/adapter/manager/CrashGridLayoutManager.kt
package lib.kalu.adapter.manager

import android.content.Context
import android.support.v7.widget.GridLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.AttributeSet

/**
 * description: fix-bug-IndexOutOfBoundsException: Inconsistency detected. Invalid view holder adapter的解决方案
 * created by kalu on 2017/3/24 15:05
 */
class CrashGridLayoutManager : GridLayoutManager {

    private val TAG = CrashGridLayoutManager::class.java.simpleName

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes) {}

    constructor(context: Context, spanCount: Int) : super(context, spanCount) {}

    constructor(context: Context, spanCount: Int, orientation: Int, reverseLayout: Boolean) : super(context, spanCount, orientation, reverseLayout) {}

    override fun onLayoutChildren(recycler: RecyclerView.Recycler?, state: RecyclerView.State) {
        try {
            super.onLayoutChildren(recycler, state)
        } catch (e: Exception) {
            //   LogUtil.e(TAG, e.getMessage(), e);
        }

    }
}
